iothub/device/src/Transport/Amqp/AmqpUnit.cs

```diff
@@ -196,23 +206,12 @@ internal async Task<AmqpIotSession> EnsureSessionIsOpenAsync(CancellationToken c
             return _amqpIotSession;
         }
 
-        public async Task CloseAsync(CancellationToken cancellationToken)
+        internal async Task CloseAsync(CancellationToken cancellationToken)
         {
             if (Logging.IsEnabled)
                 Logging.Enter(this, nameof(CloseAsync));
 
-            try
```

Comment:
We don't put try/catch/throw around semaphore waits in other places, so removing it here too, for consistency.